### PR TITLE
feat(server): add distinct tools for active vs available modules

### PR DESCRIPTION
**Pull request by carlosmmatos**
## Summary

Resolves confusion around module visibility by implementing a clear two-tool approach that separates "what modules are currently active" from "what modules are available for configuration." 

This started because @luckb0x and I were testing ADK and noticed that even if we provided `--modules` it would list all modules. This makes it clearer to the end user.

### Core Changes
- **Added** `falcon_list_enabled_modules` - Lists currently enabled/running modules based on `--modules` configuration
- **Change** `falcon_list_modules` - Lists all discoverable modules that can be enabled

### Testing & Quality
- Added 4 new comprehensive test cases covering both tools and edge cases
- All 8 tests passing with maintained code quality
- Updated tool count and method implementations

### Documentation
- Updated README.md Core Functionality section to document both tools
- Clear descriptions:"List enabled modules" vs "List available modules"

## User Value

- **Clear Intent**: Tool names make purpose immediately obvious
- **Complete Coverage**: Users can answer both "what do I have?" and "what could I have?"
- **AI Assistant Friendly**: AI can suggest additional modules based on available options
- **Configuration Discovery**: Users can explore modules without restarting or checking docs

## Example Behavior

When running `falcon-mcp --modules detections,cloud`:

- `falcon_list_enabled_modules()` → `{"modules": ["detections", "cloud"]}`
- `falcon_list_modules()` → `{"modules": ["detections", "cloud", "hosts", "idp", "incidents", "intel", "spotlight"]}`

## Testing

```bash
pytest tests/test_server.py -v  # All 8 tests pass
```

### CODE REVIEW

File: falcon_mcp/server.py [hunk 1]
```diff
@@ -113,13 +113,19 @@ def _register_tools(self) -> int:
             description="Check connectivity to the Falcon API.",
         )
 
+        self.server.add_tool(
+            self.get_active_modules,
+            name="falcon_get_active_modules",
+            description="Show active modules.",
```
[the remark this answers is not in the document]
Comment (cristianmessel-cs) [reply falcon_mcp/server.py:119]: enabled/available
active/available

Both work in my opinion, but since this is part of the description of the tool, I do agree some more details might make sense. I wouldn't however add the ENV variable because the LLM doesn't "know" about it.

Could be something like:

`Retrieves/Shows/Gets the currently active/enabled modules in the MCP/falcon-mcp server.`

Or even maybe without the word `currently`

**Comment by carlosmmatos** [comment]
@protiumx | @cristianmessel-cs  - I've updated the wording around this. Let me know what you think.

**Review by cristianmessel-cs — APPROVED** [review]
